examples/svm/plot_svm_kernels.py

```diff
@@ -1,93 +1,306 @@
+legend_labels = ["class 0", "class 1"]
+legend_handles = [
+    plt.Line2D([0], [0], marker="o", color="w", markerfacecolor=color, markersize=10)
+    for color in samples_colormap
+]
+ax.legend(legend_handles, legend_labels)
+ax.set_title("Samples in two-dimensional feature space")
+_ = plt.show()
+
+
+# %%
+# We can see that the samples are not clearly separable by a straight line.
+#
+# Training SVC model and plotting decision boundaries
+# ---------------------------------------------------
+# We create a function to plot the decision boundaries learned by the trained
+# :class:`~sklearn.svm.SVC` model, allowing the `kernel` parameter as input.
+#
+# First, we train the model.
+#
+# Next, we obtain the decision boundary by using the fitted classifier's
+# :func:`~sklearn.svm.SVC.decision_function`. In a prediction task, this
+# function computes the signed distance of each new sample to the hyperplane
+# (decision boundary) to determine its class. Here, we use it to get the
+# position of the hyperplane.
+#
+# We then color the areas separated by the hyperplane based on their predicted
+# class. We also plot the margins to both sides of the decision boundary and the
+# support vectors used during training.
+#
+# Please note that in this example, for the sake of simplicity, we do not fine
+# tune any of the :class:`~sklearn.svm.SVC` parameters. The `C` parameter is set
+# to its default value (`C=1`), resulting in margins at 1 unit distance from
+# both sides of the hyperplane. The `gamma` parameter is set to a constant value
+# (`gamma=2`) for simplicity across all kernels, although it is automatically
+# ignored for the linear kernel.
+#
+# In a real classification task, where performance matters, parameter tuning
+# with techniques like :class:`~sklearn.model_selection.GridSearchCV` is
+# highly recommended to capture different structures within the data.
+from matplotlib.colors import ListedColormap
+
+from sklearn import svm
+
+
+def plot_training_data_with_decision_boundary(kernel):
+    # Train the SVC
+    clf = svm.SVC(kernel=kernel, gamma=2)
+    clf.fit(X, y)
 
-    plt.axis("tight")
+    # Plotting settings
+    fig, ax = plt.subplots(figsize=(4, 3))
+    ax.grid(True)
     x_min = -3
     x_max = 3
     y_min = -3
     y_max = 3
+    plt.xlim(x_min, x_max)
+    plt.ylim(y_min, y_max)
+    samples_colormap = np.array(["b", "r"])
 
+    # Extract the decision boundary
     XX, YY = np.mgrid[x_min:x_max:200j, y_min:y_max:200j]
     Z = clf.decision_function(np.c_[XX.ravel(), YY.ravel()])
 
     # Put the result into a color plot
     Z = Z.reshape(XX.shape)
-    plt.figure(fignum, figsize=(4, 3))
-    plt.pcolormesh(XX, YY, Z > 0, cmap=plt.cm.Paired)
+    plt.pcolormesh(XX, YY, Z > 0, cmap=ListedColormap(["lightblue", "lightpink"]))
     plt.contour(
         XX,
         YY,
         Z,
         colors=["k", "k", "k"],
         linestyles=["--", "-", "--"],
-        levels=[-0.5, 0, 0.5],
+        levels=[-1, 0, 1],
     )
```

Comment:
We can use our own plotting tools here. At the end it will not be mush different but it hide the `meshgrid`:

```python
       from sklearn.inspection import DecisionBoundaryDisplay

    common_params = {
        "estimator": clf, "X": X, "ax": ax, "response_method": "decision_function"
    }
    DecisionBoundaryDisplay.from_estimator(**common_params, plot_method="pcolormesh")
    DecisionBoundaryDisplay.from_estimator(
        **common_params,
        plot_method="contour",
        levels=[-1, 0, 1],
        colors=["k", "k", "k"],
        linestyles=["--", "-", "--"],
    )
```